pkg/controllers/monitoring/thanos-querier/components.go

```diff
@@ -65,6 +95,12 @@ func newThanosQuerierDeployment(name string, spec *msoapi.ThanosQuerier, sidecar
 					Containers: []corev1.Container{
 						{
 							Name:  "thanos-querier",
+							Env: []corev1.EnvVar{
+								{
+									Name: "TLS_CONFIG_HASH",
+									Value: tlsHash,
+								},
+							},
```

Comment:
In CMO we use the annotations for the same purpose. Naming is along the lines of `monitoring.openshift.io/<secre-name>-hash`. Is there any reason to prefer an env var here?

I think the annotation has two advantages: A more explicit name. We could use `monitoring.rhobs/<secret-name>-hash`. And future readers will spend no time wondering what thanos might do with the mysterious env var :)

Reply:
I was wondering how the rest of the world did this :smile: . There is no particular reason to use env vars over annotations. It's just what we use in openstack-k8s-operators, so it's what I'm used to. I'll redo the code to use annotations instead.

Reply:
I rewrote the code to use annotations as you suggested. I ended up using a bit different name though if that's ok. The format I used is `monitoring.openshift.io/<secret-name>-<secret-key>-hash`.

If we used only `monitoring.openshift.io/<secret-name>-hash`, we would have issues for example when the certificate and the certificate key are in a single secret, but under different keys (which I think will happen most of the time). If we compute the hash only from the keys used for the TLS as it's doing right now and then try to save both into the same annotation, we will overwrite one of them and thus we won't be able to react if the data from the overwritten key changes and the other doesn't.

Other solutions might be:
- Always compute hashes from the whole secret (all of the keys)
- Add a code to determine from which keys to compute the hashes for each secret
- Just name the annotations differently altogether. Like `monitoring.openshift.io/tls-ca-cert-hash`, `monitoring.openshift.io/tls-key-hash` `monitoring.openshift.io/tls-ca-cert-hash`
- Use one annotation for all three values together. The data could be appended and a single hash would be computed.